### PR TITLE
Add a FreeBSD build to CI

### PR DESCRIPTION
**Pull request by zanieb**
Playing with this because it's interesting and I learned about this cool firecracker action.

Related #3370

### CODE REVIEW

File: crates/uv-performance-flate2-backend/Cargo.toml [hunk 1]
```diff
@@ -6,8 +6,8 @@ publish = false
 [lib]
 doctest = false
 
-[target.'cfg(not(any(target_arch = "s390x", target_arch = "powerpc64")))'.dependencies]
+[target.'cfg(not(any(target_arch = "s390x", target_arch = "powerpc64", target_os = "freebsd")))'.dependencies]
```
Comment (zanieb) [inline crates/uv-performance-flate2-backend/Cargo.toml:9]: This is a real behavioral change. I'm not sure if it's needed unless you're cross-building like I am? Seems fine though? Should I split into a separate PR for changelog?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -551,6 +551,50 @@ jobs:
...
+        with:
+          verbose: false
+          pre-run: |
+            include_path="$(mktemp)"
+            cat <<EOF > $include_path
+            target
+            target/x86_64-unknown-freebsd
+            target/x86_64-unknown-freebsd/debug
+            target/x86_64-unknown-freebsd/debug/uv
+            EOF
```
Comment (konstin) [inline .github/workflows/ci.yml:585]: Can you add a comment why we're listing more than one path here? It's not apparent to me from the rsync man page

**Review by konstin — APPROVED** [review]
That firefox vm is cool

File: .github/workflows/ci.yml (hunk 2 above)
Comment (zanieb) [reply .github/workflows/ci.yml:585]: I don't understand either. I had to copy this from rbspy — it didn't work with just the last path. I guess the exclude `*` overrides the includes unless we provide each directory?

Comment (konstin) [reply .github/workflows/ci.yml:587]: just that as inline comment would be helpful for when we need to touch that job again
